Data/ByteString/Lazy/Internal.hs

```diff
@@ -275,6 +276,19 @@ concat css0 = to css0
     to []               = Empty
     to (cs:css)         = go cs css
 
+-- | Repeats the given ByteString n times.
+times :: Integral a => a -> ByteString -> ByteString
+times 0 _ = Empty
+times n Empty
+  | n < 0 = error "stimes: non-negative multiplier expected"
+  | otherwise = Empty
+times n lbs0@(Chunk bs lbs)
+  | n < 0 = error "stimes: non-negative multiplier expected"
```

Comment:
The duplication of the `error` case is rather inelegant and confusing. Please handle it only once, for example by scrutinizing the `ByteString` in a separate `case`-expression.

Reply:
Yup, looks much better now